xapian-letor/api/featuremanager_internal.cc

```diff
@@ -89,16 +90,19 @@ FeatureManager::Internal::create_rank_list(const Xapian::MSet & mset, std::strin
         double weight = i.get_weight();
 
         map<int,double> fVals = transform(doc, weight);
```

Comment:
`transform()` is the method which will benefit the most from `feature_type` introduction and also the method names `calculate_fx()` therein.